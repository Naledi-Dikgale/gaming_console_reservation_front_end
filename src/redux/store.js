import { configureStore } from '@reduxjs/toolkit';
import userReducer from './userSlice';
import reservationsReducer from './reservations/reservationsSlice';
import consoleReducer from './consoleSlice';
import reserveConsoleReducer from './reservations/reserveConsoleSlice';

const store = configureStore({
  reducer: {
    user: userReducer,
    reservations: reservationsReducer,
    reserveConsole: reserveConsoleReducer,
    consoles: consoleReducer,
  },
});

export default store;
