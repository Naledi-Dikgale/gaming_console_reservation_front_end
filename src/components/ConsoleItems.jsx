import { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { getConsoles } from '../redux/consoleSlice';
import { CiFacebook, CiTwitter, CiInstagram } from 'react-icons/ci';
import { Swiper, SwiperSlide } from 'swiper/react';
import { Link } from 'react-router-dom';

import 'swiper/css';
import 'swiper/css/navigation';
import 'swiper/css/pagination';
import 'swiper/css/scrollbar';

import SliderNextButton from './SliderNextButton';
import SliderPrevButton from './SliderPrevButton';

const ConsoleItems = () => {
  const dispatch = useDispatch();
  const consoles = useSelector((state) => state.consoles.consoles);

  const [slidesPerView, setSlidesPerView] = useState(3);

  const handleResize = () => {
    if (window.innerWidth < 640) {
      setSlidesPerView(1);
    } else if (window.innerWidth >= 640 && window.innerWidth < 768) {
      setSlidesPerView(2);
    } else if (window.innerWidth >= 768) {
      setSlidesPerView(3);
    }
  };

  useEffect(() => {
    handleResize();
    window.addEventListener('resize', handleResize);
    return () => {
      window.removeEventListener('resize', handleResize);
    };
  }, []);

  useEffect(() => {
    dispatch(getConsoles())
  }, [dispatch]);

  return (
    <div className="flex flex-col w-full justify-around h-screen relative ">
      <div className="w-full flex flex-col items-center">
        <h1 className="text-xl font-extrabold">LATEST CONSOLES</h1>
        <p className="text-xs">Please select a Console</p>
      </div>

      <Swiper
        spaceBetween={10}
        slidesPerView={slidesPerView}
        onSlideChange={() => console.log('slide change')}
        onSwiper={(swiper) => console.log(swiper)}
        className="static w-full flex justify-around px-9"
      >
        {consoles.map((console) => (
          <SwiperSlide key={console.id} className="flex flex-col gap-1 w-60 h-72 items-center justify-between p-2">
            <img src={console.photo} alt={console.name} className="w-4/6 h-2/5" />
            <Link to={`/console/${console.id}`}>

              <p className="text-lg font-bold">{console.name}</p>
            </Link>
            <div className="flex flex-col gap-1 items-center">
              <p className="text-center text-xs px-3">{console.description.slice(0, 60)}...</p>
              <div className="flex gap-3 text-lg">
                <CiFacebook />
                <CiTwitter />
                <CiInstagram />
              </div>
            </div>
          </SwiperSlide>
        ))}

        <SliderPrevButton />
        <SliderNextButton />

      </Swiper>
    </div>
  )
}
export default ConsoleItems;
