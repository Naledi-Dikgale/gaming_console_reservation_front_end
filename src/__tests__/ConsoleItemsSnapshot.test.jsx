import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import { describe, test, expect } from 'vitest';
import ConsoleItems from '../components/consoles/ConsoleItems';

describe('Consoles', () => {
  test('should render consoles', () => {
    const consoles = render(
      <Provider store={store}>
        <BrowserRouter>
          <ConsoleItems />
        </BrowserRouter>
      </Provider>
    );

    expect(consoles).toMatchSnapshot();
  });
});
