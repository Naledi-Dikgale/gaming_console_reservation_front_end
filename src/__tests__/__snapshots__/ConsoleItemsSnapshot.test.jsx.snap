// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Consoles > should render consoles 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col w-full justify-around h-screen relative "
      >
        <div
          class="w-full flex flex-col items-center"
        >
          <h1
            class="text-4xl font-bold"
          >
            LATEST CONSOLES
          </h1>
          <p
            class="text-lg"
          >
            Please select a Console
          </p>
        </div>
        <div
          class="swiper swiper-initialized swiper-horizontal static w-full flex justify-around px-9"
        >
          <div
            class="swiper-wrapper"
            style="transition-duration: 0ms; transition-delay: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
          />
          <button
            class="absolute top-1/2 left-0 transform -translate-y-1/2 z-10 bg-gray-200 text-white flex justify-end items-center md:w-12 lg:w-20 py-3 pr-4 rounded-r-full hover:bg-lime-400"
          >
            <span
              class="transform rotate-180"
            >
              <svg
                class="w-6 h-6"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.25 5.653c0-.856.917-1.398 1.667-.986l11.54 6.348a1.125 1.125 0 010 1.971l-11.54 6.347a1.125 1.125 0 01-1.667-.985V5.653z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </span>
          </button>
          <button
            class="absolute top-1/2 right-0 transform -translate-y-1/2 z-10 bg-gray-200 text-white flex justify-start items-center md:w-12 lg:w-20 py-3 pl-4 rounded-l-full hover:bg-lime-400"
          >
            <span>
              <svg
                class="w-4 h-6"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.25 5.653c0-.856.917-1.398 1.667-.986l11.54 6.348a1.125 1.125 0 010 1.971l-11.54 6.347a1.125 1.125 0 01-1.667-.985V5.653z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col w-full justify-around h-screen relative "
    >
      <div
        class="w-full flex flex-col items-center"
      >
        <h1
          class="text-4xl font-bold"
        >
          LATEST CONSOLES
        </h1>
        <p
          class="text-lg"
        >
          Please select a Console
        </p>
      </div>
      <div
        class="swiper swiper-initialized swiper-horizontal static w-full flex justify-around px-9"
      >
        <div
          class="swiper-wrapper"
          style="transition-duration: 0ms; transition-delay: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
        />
        <button
          class="absolute top-1/2 left-0 transform -translate-y-1/2 z-10 bg-gray-200 text-white flex justify-end items-center md:w-12 lg:w-20 py-3 pr-4 rounded-r-full hover:bg-lime-400"
        >
          <span
            class="transform rotate-180"
          >
            <svg
              class="w-6 h-6"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.25 5.653c0-.856.917-1.398 1.667-.986l11.54 6.348a1.125 1.125 0 010 1.971l-11.54 6.347a1.125 1.125 0 01-1.667-.985V5.653z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
        </button>
        <button
          class="absolute top-1/2 right-0 transform -translate-y-1/2 z-10 bg-gray-200 text-white flex justify-start items-center md:w-12 lg:w-20 py-3 pl-4 rounded-l-full hover:bg-lime-400"
        >
          <span>
            <svg
              class="w-4 h-6"
              fill="none"
              stroke="currentColor"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.25 5.653c0-.856.917-1.398 1.667-.986l11.54 6.348a1.125 1.125 0 010 1.971l-11.54 6.347a1.125 1.125 0 01-1.667-.985V5.653z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
